libs/core/src/lib/decorators/property.ts

```diff
@@ -32,25 +32,77 @@ export interface IPropertyOptions
     maxValue?: number;
 
     /**
-     * True if property is read only.
+     * True if property is read only in Intuiface Composer. The property will only be usable as a binding source.
      */
     readOnly?: boolean;
 
     /**
-     * The type of the property
+     * The type of the property.
+     * It's optional but higly recommended fo better binding conversion.
+     * It can be:
+     * - `String`
+     * - `Number`
+     * - `Boolean`
+     * - `Array`
+     * - Classe name of a {@link IConvertibleType | convertible type} (`Color`, `Font`, `Path`, `Resource`, `Time`)
+     * - Enum type
      */
     type?: any;
 
     /**
-     * The item type in case the type is array
+     * The item type in case the type is `Array`.
+     * It's optional but useful in interface assets to display item's properties in Intuiface Composer.
+     * It can be:
+     *  - Class name
+     *  - `String`
+     *  - `Number`
+     *  - `Boolean`
      */
     itemType?: any;
 }
 
 /**
- * Property decorator to add metadatas to property.
+ * The `@Property` decorator enable you to declare a Property on your asset that can be used in a Intuiface experience.
  * @param options options of the property (display name, description, ...)
+ *
+ * @example ```ts
+ * @Property({
+ *     displayName: 'Volume', // 'Volume' is the name of the property
+ *     description: 'Current volume in the media.', // here the description of the property
+ *     defaultValue: 1, // the default value of the property : 1
+ *     minValue: 0, // the minimum value : 0
+ *     maxValue: 1, // the maximum value : 1
+ *     type: Number // the property is a number (so binding on a text with '0.5' value will be converted in a number value 0.5)
+ * })
+ * public volume: number = 0; // declaration of the property
+ * ```
+ * _**Note**_: the name `volume` is in camelCase as the naming convention.
+ *
+ * ❗⚠️⚠️⚠️⚠️❗For property type `Array` there is a limitation: if you modofy the array with methods like `push`, `pop`, `reduce`, `reverse`, `shift`, `sort`, `splice`... without calling a setter (e.g. `this.myArray = [...]`) bindings will not be updated. To fix that, you can use the method {@link Watchable.notifyPropertyChanged}.
+ * @example I have an itme list declared like this:
+ * ```ts
+ * /**
+ *  * Item List
+ *  *\/
+ * @Property({
+ *     displayName: 'List Items',
+ *     defaultValue: [],
+ *     type: Array,
+ *     itemType: ListItem
+ * })
+ * public listItems: ListItem[] = [];
+ * ```
+ * I have an action which adds an item to the list using the `push` method. I have to add this code to be sure all my bindings will be resolved when I add a new item:
+ * ```ts
+ * // add my new item to the list
+ * this.listItems.push(newItem);
+ * // call the notify property changed
+ * this.notifyPropertyChanged('listItems', this.listItems);
+ * ```
+ *
+ * @group Decorators
  */
+// eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
Maybe we can use https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#custom to allow for example `export function` and the name following in PascalCase. 
This will avoid to disable eslint rule for each decorator (maybe an evolution to do later)

Reply:
Ok. For later.